src/main/java/edu/harvard/iq/dataverse/authorization/AuthenticationServiceBean.java

```diff
@@ -380,14 +380,7 @@ public ApiToken findApiTokenByUser(AuthenticatedUser au) {
         }
         if (tokens.size() == 1) {
             // Normal case - one token that may or may not have expired
-            ApiToken token = tokens.get(0);
-            if (token.getExpireTime().before(latest)) {
-                // Don't return an expired token which is unusable, delete it instead
-                em.remove(token);
-                return null;
-            } else {
-                return tokens.get(0);
-            }
+            return tokens.get(0);
         } else {
             // We have more than one due to https://github.com/IQSS/dataverse/issues/6389 or
             // similar, so we should delete all but one token.
```

Comment:
should we also then not delete expired token in this code here (I'm a little confused by the comments here which say more that one token, but then also "if we only have an expired token" (implying one)). My guess is is still means multiple, but removed all expired, which if they all are would be the same logic we're trying to change above.

Reply:
~yeah - good catch. In theory this code only runs once, but it's here in case there are other ways to get multiple tokens aside from the original bug, so it should be consistent. I updated the code.